### ISSUE TITLE
image descriptions for SVGs (\tikzpictures)

### ISSUE THREAD

**Issue by rzach** [brucemiller/LaTeXML#2165]
Issue  #1452 added support for `alt` attributes for `img` tags produced by `\includegraphics`. It looks like no similar mechanism exists for images produced with `\tikzpicture` (which are turned into inline SVG).

**Comment by dginev** [comment]
Interesting observation, @rzach . I believe we followed the lead of the LaTeX team for the `alt` support, who added the key to the `\includegraphics` macro. Is there a similar development for the native `\tikzpicture`? 

I don't know the answer, and maybe this extra information can be requested from the [pgf-tikz](https://github.com/pgf-tikz) project? 

I am finding a similar request for [htlatex's {tikzpicture}](https://tex.stackexchange.com/questions/641335/htlatex-tikzpicture-with-alt-tag-and-custom-class), but that doesn't really indicate if this has become native since...

**Comment by rzach** [comment]
If I understand the spec correctly, `<svg>` doesn't have an `alt` attribute, but rather you put title and/or description inside the `<svg>` between `<title>...</title>` and `<desc>...</desc>` tags. Since you don't know if `latexml` will generate inline SVGs or write to an image file or whatnot, I suppose it's harder to decide how to implement.

Perhaps the easiest way to do this is to provide something in `latexml.sty` so you can insert `<title>` and `<desc>` into the SVG code (`\lxSvgTitle`, `\lxSvgDesc`, or something more general that adds raw XML?)

[The way I get around it now is to use features provided by [https://github.com/vlmantova/bookml](BookML): Surround the `tikzpicture` with `\begin{bmlimage}...\end{bmlimage}`. BookML then converts the picture to SVG by running it through LaTeX and converting the DVI to an SVG file which is then included using `<img>`. And you can set the `alt` tag on that image using `\bmlDescription{...}` right after `\end{bmlimage}`. That works well enough but the images generated aren't optimal--for one, everything is in Computer Modern. But if LaTeXML had an option to write the SVG to a file instead of inlining it (which I think is issue #2144?), and a way to provide the ALT tag on the resulting `<img>`, ...]

**Comment by dginev** [comment]
Right, sure. I am not that concerned about passing the information into the SVG, various nice [accessibility guides](https://www.tpgi.com/using-aria-enhance-svg-accessibility/) exist to that end, and it is relatively easy to accommodate the different approaches in latexml.

What is less certain - and I am fishing for - is how to build an authoring solution supported by the official tikz package (the way `alt` is now an official \includegraphics key in `graphicx.sty`), so that the same LaTeX source can be converted with any of latexml/tex4ht/pandoc/plastex/tralics/... without reinventing the markup inside each variant. 

The description has to be provided by the author of the figure, so that's a main prerequisite here, before we start fleshing out latexml implementation details.

**Comment by rzach** [comment]
This doesn't address the issue you raise (of having a way to provide title and description in the `tikzpicture` using some mechanism provided by TikZ directly), but another option would be to provide an environment in `latexml.sty` that can wrap things in a `div` with a ARIA attributes. Then authors could produce an inline SVG (or other graphic with with even just styled boxes), wrap it in a `<div aria-hidden=true>`, and wrap the whole thing in `<div aria-label="...">...</div>`. (The `aria-hidden` div might be necessary to prevent AT from reading out text embedded in the SVG.)
